src/main/kotlin/bandcampcollectiondownloader/BandcampCollectionDownloader.kt

```diff
@@ -157,6 +171,20 @@ object BandcampCollectionDownloader {
             return
         }
 
+        // Filter by Date
+        if (args.filterDate != null) {
+            try {
+                val filterDate: LocalDate = LocalDate.now();
+                val filterInstant: Instant = filterDate.atStartOfDay(ZoneId.systemDefault()).toInstant();
+            
+                if (releaseUTC != null && releaseUTC.toInstant().isAfter(filterInstant)) {
+                    return 
+                }
+            } catch(e: DateTimeParseException) {
+                Util.log("could not parse filterDate")
```

Comment:
I find it a little late for checking whether the provided argument is a formatted date or not. I would recommend making sure the date is correct as soon as when the arguments parsing occurs, using what PicoCLI offers for that (https://picocli.info/#_strongly_typed_everything). See my comment above on that part.